test/createParser.js

```diff
@@ -0,0 +1,16 @@
+var {DOMParser} = require('../lib/dom-parser')
+
+function createParser ({errorHandler, errors = {}, locator = {}} = {}) {
+  errorHandler = errorHandler || ((key, msg) => {
+    if (!errors[key]) errors[key] = []
+    errors[key].push(msg)
```

Comment:
minor nit: I think newly added code should avoid overwriting function parameters unless absolutely necessary, which I think is not the case here.

nit 2: I find the file name `createParser.js` a little too generic. If it is intended to be used by other tests in the future, I would favor calling it something like createTestParser.js. If it is only expected to be used for xmltest, I would favor reflecting that in the module name.

another minor nit: I think all other modules in `lib` and `test` are in kebab case, which I would probably favor.

Reply:
In this case it's actually nessesary to pass those parameters, tsince this way the "errors, warnings and fatals" are part of the snapshot assertion.Should one of them go missing or change the test would break.
The error handlers are part of the public API and the code to handle them the same way in multiple tests is quite verbose, to I decided to extract them.
Since it's still work in progress, I really don't bother to much with names.
I'm also not sure if we should use it outside of the `xmltest` "suite", so I'm fine with whatever name you prefer.

> kebab case

You are right! I really didn't realize that "convention" existed consistently in the `test` folder. Most modules in this repo, especially in the test folder are not really helpful with their naming, and it felt natural to name a module after the only function it exports.
But I don't have any concerns with renaming it.

Reply:
Thinking about it a bit longer: maybe I just didn't get your point about "overriding".
If my reply was not satisfying, please let me know and elaborate your concerns.

The renaming took place, please check if this is what you had in mind.